Sources/Controllers/2.0/OAMapSettingsMainScreen.mm

```diff
@@ -530,6 +545,12 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
                 [cell.switchView setOn: [_styleSettings isCategoryEnabled:@"transport"]];
                 [cell.switchView addTarget:self action:@selector(transportChanged:) forControlEvents:UIControlEventValueChanged];
             }
+            if ([data[@"key"] isEqualToString:@"wikipedia_layer"])
+            {
+                BOOL enabled = [[OAPOIFiltersHelper sharedInstance] isTopWikiFilterSelected];
+                [cell.switchView setOn:enabled];
+                [cell.switchView addTarget:self action:@selector(wikipediaChanged:) forControlEvents:UIControlEventValueChanged];
```

Comment:
Always remove target before adding in reusable contexts. (Check the entire method)

Reply:
[target has already been deleted (line 513)](https://github.com/osmandapp/OsmAnd-iOS/pull/1329/files/ddb0d33f1201038603f619142b194fabc140b135#diff-680715c853695276c3aa0c7697bda43f5edaced0f9f30f2c7866cf620fdf9583R513)